src/main/java/reactor/netty/resources/ConnectionProvider.java

```diff
@@ -158,7 +162,7 @@ static ConnectionProvider fixed(String name, int maxConnections) {
 	 * number of {@link Connection}
 	 */
 	static ConnectionProvider fixed(String name, int maxConnections, long acquireTimeout) {
-		return fixed(name, maxConnections, acquireTimeout, null, null);
```

Comment:
Keep this and invoke the builder creation just once in `fixed(String name, int maxConnections, long acquireTimeout, @Nullable Duration maxIdleTime, @Nullable Duration maxLifeTime)`